HTML translator and a web service

jerinphilip opened the pull request:
Running looks like:

```
LD_PRELOAD=/home/jerin/versioned/uedin/lemonade/env/lib/python3.8/site-packages/bergamot/_bergamot.cpython-38-x86_64-linux-gnu.so \
python3 -m tagtransfer.webapp \
	 --model-config ~/.local/share/lemonade/models/en-de-tiny/config.bergamot.yml \
	 --num-workers 4
```

View instructions in README.md to download required models and install lemonade python bindings into a virtual environment.

Commit:
Checking in linksurf

Commit:
Minor p update

Commit:
additional preproc layer for xhtml to html

Commit:
Raise in case of except for error

Commit:
Don't try to parse if empty tag transfer incomplete

Commit:
Make tag an arg

jerinphilip committed:
More cleaning so HTML going in is okay

jerinphilip committed:
More delta to print body tag completely

jerinphilip committed:
Converting to webapp

jerinphilip committed:
Monkeypatching things now...>

jelmervdl commented:
Fun observation:

```diff
diff --git a/tagtransfer/webapp.py b/tagtransfer/webapp.py
index c1fcbb399a582aef28a1eb881c67309c9ecb14f8..f139c0a00585263cc5eced42b0401332f3c6e404 100644
--- a/tagtransfer/webapp.py
+++ b/tagtransfer/webapp.py
@@ -16,7 +16,7 @@ def hello_world():
     tree = html.fromstring(document)
     head = tree.xpath('/html/head')[0]
     head.insert(1, html.fragment_fromstring('<base href="{}" target="_blank">'.format(url)))
-    return etree.tostring(tree)
+    return etree.tostring(tree, method='html')
 
 
 if __name__ == "__main__":
```

Just adding this bit fixes the wikipedia example. It looks like without `method='html'` `etree.tostring()` generates some `<div/>` elements which I only expect to work if you send it as a proper xml document, i.e. with the `<?xml version=1.0?>` bit and a proper content-type header.

jerinphilip committed:
html5, decoding and tidy before doc goes out

jerinphilip committed:
Don't drop empty tags, affects fontawesome

jelmervdl committed:
Add bypass=true option to url to skip translation

Helpful for figuring out whether its libtidy or us

jelmervdl committed:
Remove libtidy, fix encoding of entities, fix <script> being encoded with html entities.

By removing libtidy we get to see what the parser actually does itself. There's still etree in front of it to make sure it gets valid sgml at least.
